securesystemslib/signer.py

```diff
@@ -72,22 +71,18 @@ def from_dict(cls, signature_dict: Dict) -> "Signature":
         keyid = signature_dict.pop("keyid")
```

Comment:
If `keyid` is optional then you should provide a default value when calling `pop` otherwise you will get a `KeyError`:
```
> d = {"a": 3}
> d.pop("b")
KeyError: 'b'
> d.pop("b", None)
>
```

So, I will suggest instead do:
```suggestion
        keyid = signature_dict.pop("keyid", None)
```

Please make sure that you do that everywhere you are using `pop` on an optional argument.